app/models/proposal.rb

```diff
@@ -160,12 +169,30 @@ def on_rejected_entry(prev_state, event)
 
   def restart
     # Note that none of the state machine's history is stored
+    # Further note: this isn't used anywhere
```

Comment:
What do you mean? `expires_at` is checked by `#expired?`, as well as the `expired`/`unexpired`/`fresh` scopes on the ApiToken.